internal/cachettl/cachettl.go

```diff
@@ -0,0 +1,121 @@
+package cachettl
```

Comment:
I'm OK either way to be honest. I created a separate package because it is generic enough after all, and it's just a data structure so I saw no harm in having it in its own package without subbing it.